src/Features/Lsif/Generator/Program.cs

```diff
@@ -195,23 +207,23 @@ private static async Task LocateAndRegisterMSBuild(TextWriter logFile, Directory
         private static async Task GenerateWithMSBuildWorkspaceAsync(
             FileInfo solutionOrProjectFile,
             ILsifJsonWriter lsifWriter,
-            TextWriter logFile,
+            ILogger logger,
             Func<MSBuildWorkspace, CancellationToken, Task<Solution>> openAsync,
             CancellationToken cancellationToken)
         {
-            await logFile.WriteLineAsync($"Loading {solutionOrProjectFile.FullName}...");
+            logger.LogInformation($"Loading {solutionOrProjectFile.FullName}...");
 
             var solutionLoadStopwatch = Stopwatch.StartNew();
 
             using var msbuildWorkspace = MSBuildWorkspace.Create(await Composition.CreateHostServicesAsync());
-            msbuildWorkspace.WorkspaceFailed += (s, e) => logFile.WriteLine("Error while loading: " + e.Diagnostic.Message);
+            msbuildWorkspace.WorkspaceFailed += (s, e) => logger.Log(e.Diagnostic.Kind == WorkspaceDiagnosticKind.Failure ? LogLevel.Error : LogLevel.Warning, "Problem while loading: " + e.Diagnostic.Message);
 
             var solution = await openAsync(msbuildWorkspace, cancellationToken);
 
             var options = GeneratorOptions.Default;
 
-            await logFile.WriteLineAsync($"Load completed in {solutionLoadStopwatch.Elapsed.ToDisplayString()}.");
-            var lsifGenerator = Generator.CreateAndWriteCapabilitiesVertex(lsifWriter, logFile);
+            logger.LogInformation($"Load completed in {solutionLoadStopwatch.Elapsed.ToDisplayString()}.");
```

Comment:
Are any of these stopwatch times things that we want to be aggregated? If so, they could be passed as telemetry also/instead.

Perhaps a future improvement.

Reply:
Yes, we will, but for this was just step 1.